packages/core/src/shared/utilities/messages.ts

```diff
@@ -35,21 +35,32 @@ export function makeFailedWriteMessage(filename: string): string {
     return message
 }
 
-function showMessageWithItems(
-    message: string,
+export function showMessage(
     kind: 'info' | 'warn' | 'error' = 'error',
+    message: string,
     items: string[] = [],
-    useModal: boolean = false
-): Thenable<string | undefined> {
-    switch (kind) {
-        case 'info':
-            return vscode.window.showInformationMessage(message, { modal: useModal }, ...items)
-        case 'warn':
-            return vscode.window.showWarningMessage(message, { modal: useModal }, ...items)
-        case 'error':
-        default:
-            return vscode.window.showErrorMessage(message, { modal: useModal }, ...items)
+    options: vscode.MessageOptions & { telemetry?: boolean } = {},
+    metric: ToolkitShowNotification = {
+        id: 'unknown',
+        component: 'editor',
     }
+): Thenable<string | undefined> {
+    return telemetry.toolkit_showNotification.run(async (span) => {
```

Comment:
nice idea

Reply:
Yeah I need to think more about this, could you show a pseudocode example of what you have in mind? 

Meanwhile, I will push a lower-risk change for today's release.

Reply:
Ah, I think I can make that work in a followup PR.

Reply:
https://github.com/aws/aws-toolkit-vscode/pull/5776